src/types.ts

```diff
@@ -29,112 +93,150 @@ export interface CreateProposalDto {
 
 export interface VoteProposalDto {
   proposal: string; // proposal id
-  choice: number; // 1 or 2
+  choice: 1 | 2; // Yes or No
 }
 
 export interface ExecuteProposalDto {
   proposal: string; // proposal id
 }
 
-export interface Instance {
+export interface SDKInstance {
   /**
-   * Get all the list of zDAO
+   * Get all the list of zNA
+   * @returns list of zNA
    */
-  getZDAOs(): Promise<zDAO[]>;
+  listZNA(): Promise<zNA[]>;
 
   /**
-   * Get zDAO by zNA
-   * @param zNA zNA address to find zDAO
+   * Create an zDAO instance by zNA
+   * @param zNA zNA address
+   * @returns created zDAO instance
+   * @exception throw Error if zNA does not exist
    */
-  getZDAOByZNA(zNA: string): Promise<zDAO | undefined>;
+  getZDAOByZNA(zNA: zNA): Promise<ZDAOInstance>;
 
   /**
-   * Get zDAO assets by zNAs
-   * @param zNA zNA address to get zDAO Assets
+   * Check if zDAO exists
+   * @param zNA zNA address
+   * @returns true if zNA exists
    */
-  getZDAOAssetsByZNA(zNA: string): Promise<
-    | {
-        amountInUSD: number;
-        assets: Array<Asset>;
-      }
-    | undefined
-  >;
+  doesZDAOExist(zNA: zNA): Promise<boolean>;
 
   /**
-   * Get zDAO transactions by zNA
-   * @param zNA zNA address to get zDAO assets
+   * Create zDAO from parameters for test
+   * @param packaged parameters of zDAO
+   * @exception throw Error if zNA already exists
+   * @exception throw Error if owners is empty
+   * @exception throw Error if title is empty
    */
-  getZDAOTransactionsByZNA(zNA: string): Promise<Array<Transaction>>;
+  createZDAOFromParams(param: CreateZDAODto): Promise<void>;
+}
 
+export interface ZDAOInstance {
   /**
-   * Get all the proposals added in the zDAO
-   * @param zNA zNA address
+   * Get zDAO
+   * @returns zDAO structure
+   */
+  getZDAO(): zDAO;
```

Comment:
Consider just making this interface extend `zDAO`.

Or call this `getDetails()` since it would be confusing to "get the ZDAO of a ZDAO Instance"

Reply:
or `.details()`